src/wp-admin/includes/privacy-tools.php

```diff
@@ -388,14 +385,32 @@ function wp_privacy_generate_personal_data_export_file( $request_id ) {
 		),
 	);
 
-	// Merge in the special about group.
-	$groups = array_merge( array( 'about' => $about_group ), $groups );
-
-	$groups_count = count( $groups );
+	// And now, all the Groups.
+	$groups = get_post_meta( $request_id, '_export_data_grouped', true );
+	if ( false === $groups ) {
+		$groups       = array( 'about' => $about_group );
+		$groups_count = 1;
+	} else {
+		if ( ! is_array( $groups ) ) {
+			_doing_it_wrong(
+				__FUNCTION__,
+				/* translators: %s: post meta key. */
+				sprintf( __( 'The %s post meta must be an array.' ), "<code>'_export_data_grouped'</code>" ),
+				'5.8.0'
+			);
+			$groups = (array) $groups;
+		}
+		$groups       = array_merge( array( 'about' => $about_group ), $groups );
+		$groups_count = count( $groups );
+	}
```

Comment:
When post meta exists and is not an array, type casting and then merging causes a failure downstream. Why? The groups meta will not be in the right format as keys are not set. In this case, the problem is caused from here. The merging of non array means the post meta is ignored in < PHP 8. Therefore, skipping the type casting and merge when not an array is valid here.

https://gist.github.com/jrfnl/5f16fcc747c96339fa7133e0ac086e2d

Reply:
Resolved by commit https://github.com/WordPress/wordpress-develop/pull/758/commits/ae220090decbd37d30493d0500520b47214eb80c